arm-servicefabric/2016-09-01/swagger/servicefabric.json

```diff
@@ -0,0 +1,939 @@
+          "type": "integer"
+        },
+        "maxPercentUnhealthyApplications": {
+          "type": "integer"
+        },
+        "applicationHealthPolicies": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/PaasApplicationHealthPolicy"
+          }
+        }
+      }
+    },
+    "PaasServiceTypeDeltaHealthPolicy": {
+      "description": "Paas service type delta health policy",
+      "properties": {
+        "maxPercentDeltaUnhealthyServices": {
+          "type": "integer"
+        }
+      }
+    },
+    "PaasApplicationDeltaHealthPolicy": {
+      "description": "Paas application delta health policy",
+      "properties": {
+        "defaultServiceTypeDeltaHealthPolicy": {
+          "$ref": "#/definitions/PaasServiceTypeDeltaHealthPolicy"
+        },
+        "serivceTypeDeltaHealthPolicies": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/PaasServiceTypeDeltaHealthPolicy"
+          }
+        }
+      }
+    },
+    "PaasClusterUpgradeDeltaHealthPolicy": {
+      "description": "Paas cluster upgrade delta health policy",
+      "properties": {
+        "maxPercentDeltaUnhealthyNodes": {
+          "type": "integer"
+        },
+        "maxPercentUpgradeDomainDeltaUnhealthyNodes": {
+          "type": "integer"
+        },
+        "maxPercentDeltaUnhealthyApplications": {
+          "type": "integer"
+        },
+        "applicationHealthPolicies": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/PaasApplicationDeltaHealthPolicy"
+          }
+        }
+      }
+    },
+    "PaasClusterUpgradePolicy": {
+      "description": "Paas cluster upgrade policy",
+      "properties": {
+        "overrideUserUpgradePolicy": {
+          "type": "boolean"
+        },
+        "forceRestart": {
+          "type": "boolean"
+        },
+        "upgradeReplicaSetCheckTimeout": {
+          "type": "string"
+        },
+        "healthCheckWaitDuration": {
+          "type": "string"
+        },
+        "healthCheckStableDuration": {
+          "type": "string"
+        },
+        "healthCheckRetryTimeout": {
+          "type": "string"
+        },
+        "upgradeTimeout": {
+          "type": "string"
+        },
+        "upgradeDomainTimeout": {
+          "type": "string"
+        },
+        "healthPolicy": {
+          "$ref": "#/definitions/PaasClusterUpgradeHealthPolicy"
+        },
+        "deltaHealthPolicy": {
+          "$ref": "#/definitions/PaasClusterUpgradeDeltaHealthPolicy"
+        }
+      }
+    },
+    "DiagnosticsStorageAccountConfig": {
+      "description": "Diagnostics storage account config",
+      "properties": {
+        "storageAccountName": {
+          "type": "string"
+        },
+        "primaryAccessKey": {
+          "type": "string"
+        },
+        "secondaryAccessKey": {
+          "type": "string"
+        },
```

Comment:
When an ARM resource is referenced, it should be done using the fully qualified resource id. This means here the storage account should be referenced using fully qualified arm resource id. Your RP can enable linked access check on the storage account. This way RP can enumerate the storage account credentials and endpoints.

Please log a bug on your system to track this.